src/runtime_src/core/edge/sw_emu/generic_pcie_hal2/shim.cxx

```diff
@@ -998,11 +1000,25 @@ namespace xclswemuhal2 {
         }
       }
     }
+    // TCP socket thread sets a promise to ensure it is started.
+    // once, thread starts, server gets ready by shim driver
+    // device process can easily make connect() call when server is ready.
+    auto fut = tcp_socket_pr.get_future();
     //Create thread for TCP socket connection
     std::thread tcpSockThread = std::thread(&SwEmuShim::socketConnection, this, true);
 
     bool simDontRun = xclemulation::config::getInstance()->isDontRun();
     if (!simDontRun) {
+       std::future_status status;
+       auto counter = 0;
+      do {
+        status = fut.wait_for(std::chrono::seconds(1));
+      } while (status != std::future_status::ready && ++counter < 30);
+      
+      if (status != std::future_status::ready) {
+        std::cerr<<"ERROR: [SW_EMU 13] TCP/IP socket thread is not started yet,something wrong with OS, So exiting the application now.";
+        exit(1);
```

Comment:
return a failure instead of exiting. Some other places also using exit, but as a library, returning a failure to upper stack is the right practice..

Reply:
Agreed. Core layer will intimate the failure. Modified as per the comment.